Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -286,6 +337,26 @@ public protocol ChatChannelListControllerDelegate: DataControllerStateDelegate {
         _ controller: ChatChannelListController,
         didChangeChannels changes: [ListChange<ChatChannel>]
     )
+    
+    /// The controller asks the delegate if the newly inserted `ChatChannel` should be linked to this Controller's query.
+    /// Defaults to `false`
+    /// - Parameters:
+    ///   - controller: The controller,
+    ///   - shouldLinkNewChannel: The newly inserted `ChatChannel` instance. This instance is not linked to the controller's query.
+    func controller(
+        _ controller: ChatChannelListController,
+        shouldLinkNewChannel: ChatChannel
+    ) -> Bool
+    
+    /// The controller asks the delegate if the newly updated `ChatChannel` should be linked to this Controller's query.
+    /// Defaults to `false`
+    /// - Parameters:
+    ///   - controller: The controller,
+    ///   - shouldLinkUpdatedChannel: The newly updated `ChatChannel` instance. This instance is not linked to the controller's query.
+    func controller(
+        _ controller: ChatChannelListController,
+        shouldLinkUpdatedChannel: ChatChannel
+    ) -> Bool
```

Comment:
I like it 👍 What if updated channel no longer matches the query, though? 🤔 Do we need:
```swift
func controller( _ controller: ChatChannelListController, shouldUnlinkUpdatedChannel: ChatChannel) -> Bool
```

Reply:
One extra came to my mind: what if we give a `completion: (Bool) -> Void` to these methods? 🙂 Then we'll also cover use-cases when it's not possible to make a decision synchronously

Reply:
users should keep the processing to a minimum, we don't want them to make an API call for this, it'd just make stuff more complex

`shouldUnlinkUpdatedChannel` is not needed for now